src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/AttributeComponent.scala

```diff
@@ -24,12 +29,114 @@ trait AttributeComponent {
     def name = column[String]("name")
     def valueString = column[Option[String]]("value_string")
     def valueNumber = column[Option[Double]]("value_number")
-    def valueDate = column[Option[Timestamp]]("value_date")
     def valueBoolean = column[Option[Boolean]]("value_boolean")
+    def valueEntityRef = column[Option[Long]]("value_entity_ref")
     def listIndex = column[Option[Int]]("list_index")
 
-    def * = (id, name, valueString, valueNumber, valueDate, valueBoolean, listIndex) <> (AttributeRecord.tupled, AttributeRecord.unapply)
+    def * = (id, name, valueString, valueNumber, valueBoolean, valueEntityRef, listIndex) <> (AttributeRecord.tupled, AttributeRecord.unapply)
+
+    def workspace = foreignKey("FK_ATTRIBUTE_ENTITY_REF", valueEntityRef, entityQuery)(_.id.?)
   }
 
-  protected val attributeQuery = TableQuery[AttributeTable]
+  protected object attributeQuery extends TableQuery(new AttributeTable(_)) {
+    def insertAttributeRecords(name: String, attribute: Attribute, workspaceId: UUID) = {
+      attribute match {
+        case AttributeEntityReferenceList(refs) => refs.zipWithIndex.map { case (ref, index) => insertAttributeRef(name, workspaceId, ref, Option(index))}
+        case AttributeValueList(values) =>
+          assertConsistentListMembers(values)
+          values.zipWithIndex.map { case (value, index) => insertAttributeValue(name, value, Option(index))}
+        case value: AttributeValue => Seq(insertAttributeValue(name, value))
+        case ref: AttributeEntityReference => Seq(insertAttributeRef(name, workspaceId, ref))
+        case AttributeEmptyList => Seq(insertAttributeValue(name, AttributeNull, Option(-1))) // storing empty list as an element with index -1
+      }
+    }
+
+    private def insertAttributeRef(name: String, workspaceId: UUID, ref: AttributeEntityReference, listIndex: Option[Int] = None) = {
+      findEntityByName(workspaceId, ref.entityType, ref.entityName).result.flatMap {
+        case Seq() => throw new RawlsException(s"$ref not found in workspace $workspaceId")
+        case Seq(entityRecord) => (attributeQuery returning attributeQuery.map(_.id)) += marshalAttributeEntityReference(name, listIndex, entityRecord)
+        case _ => throw new RawlsException(s"more than one $ref found in workspace $workspaceId")
```

Comment:
while you and I know this, scala does not, I thought the catch all would make scala happier